aws/resource_aws_db_instance.go

```diff
@@ -535,11 +535,12 @@ func resourceAwsDbInstanceCreate(d *schema.ResourceData, meta interface{}) error
 			opts.Iops = aws.Int64(int64(attr.(int)))
 		}
 
+		if arnParts := strings.Split(v.(string), ":"); len(arnParts) >= 4 {
+			opts.SourceRegion = aws.String(arnParts[3])
+		}
```

Comment:
This check would be safer and easier to understand using the AWS Go SDK provided ARN handlers [`arn.IsARN()`](https://github.com/aws/aws-sdk-go/blob/9c9db07028aeb7d607e54b2b5aa97bbbbebcbbfd/aws/arn/arn.go#L78-L82) and [`arn.Parse()`](https://docs.aws.amazon.com/sdk-for-go/api/aws/arn/#Parse), e.g.

```go
		if arn.IsARN(v.(string)) {
			replicaSourceDbArn, err := arn.Parse(v.(string))
			if err != nil {
				return fmt.Errorf("error parsing replica_source_db as ARN: %s", err)
			}
			opts.SourceRegion = aws.String(replicaSourceDbArn.Region)
		}
```

Reply:
Okay, I can make that change. Just shows how long this pr has been sitting here. (;